maint/UpdateAlways

```diff
@@ -162,21 +154,22 @@ perl <<"END" >pcre2demo.3
     print OUT;
     }
   print OUT ".EE\n";
-  close(IN);
   close(OUT);
+  close(IN);
 END
 if [ $? != 0 ] ; then exit 1; fi
 
 
 # Verify that `man` can process the pages without warnings.
-
-for file in *.1 *.3 ; do
-  MAN_OUT=`MANROFFSEQ='' MANWIDTH=80 man --warnings=w,all -E UTF-8 -l -Tutf8 -Z "$file" 2>&1 >/dev/null`
-  if [ "$MAN_OUT" != "" ]; then
-    printf "Running man generated warnings:\n%s\n" "$MAN_OUT"
-    exit 1
-  fi
-done
+if man --help 2>/dev/null | grep -q warnings ; then
+  for file in *.1 *.3 ; do
+    MAN_OUT=`MANROFFSEQ='' MANWIDTH=80 man --warnings=w,all -E UTF-8 -l -Tutf8 -Z "$file" 2>&1 >/dev/null`
+    if [ "$MAN_OUT" != "" ]; then
+      printf "Running man generated warnings:\n%s\n" "$MAN_OUT"
+      exit 1
+    fi
+  done
+fi
```

Comment:
I'd put an `else echo "Skipping man page check (non-GNU system)";` at the end, so that we don't accidentally silently skip this check if some bitrot occurs (eg if `man --help` ever changes in future). Just to increase the chance we notice.